ha/setup/v2/ha_setup.py

```diff
@@ -0,0 +1,233 @@
+        machine_id, err, rc = self._execute.run_cmd(command, check_error=True)
+        Log.info(f"Read machine-id. Output: {machine_id}, Err: {err}, RC: {rc}")
+        node_name = Conf.get(self._index, f"cluster.server_nodes.{machine_id.strip()}")
+
+        # Read cluster name and cluster user
+        cluster_name = Conf.get(self._index, 'corosync-pacemaker.cluster_name')
+        cluster_user = Conf.get(self._index, 'corosync-pacemaker.user')
+
+        # Read cluster user password and decrypt the same
+        cluster_id = Conf.get(self._index, 'cluster.cluster_id')
+        cluster_secret = Conf.get(self._index, 'corosync-pacemaker.secret')
+        key = Cipher.generate_key(cluster_id, cluster_user)
+        cluster_secret = Cipher.decrypt(key, cluster_secret.encode('ascii')).decode()
+
+        # TBD: Call script for creating and configuring the cluster
+        # TBD: Check if the cluster exists already
+
+class InitCmd(Cmd):
+
+    """Init Setup Cmd"""
+    name = "init"
+
+    def __init__(self, args):
+        """Init method."""
+        super().__init__(args)
+
+    def process(self):
+        pass # TBD
+
+class TestCmd(Cmd):
+
+    """Test Cmd"""
+    name = "test"
+
+    def __init__(self, args):
+        """Init method."""
+        super().__init__(args)
+
+    def process(self):
+        pass # TBD: Call script to check that all resources have been configured
+
+class UpgradeCmd(Cmd):
+
+    """Upgrade Cmd"""
+    name = "upgrade"
+
+    def __init__(self, args):
+        """Init method."""
+        super().__init__(args)
+
+    def process(self):
+        pass # TBD
+
+class ResetCmd(Cmd):
+
+    """Reset Cmd"""
+    name = "reset"
+
+    def __init__(self, args):
+        """Init method."""
+        super().__init__(args)
+
+    def process(self):
+        pass # TBD
+
+class CleanupCmd(Cmd):
+
+    """Cleanup Cmd"""
+    name = "cleanup"
+
+    def __init__(self, args):
+        """Init method."""
+        super().__init__(args)
+
+    def process(self):
+        pass # TBD: Call cluster destroy script here
+
+class BackupCmd(Cmd):
+
+    """Backup Cmd"""
+    name = "backup"
+
+    def __init__(self, args):
+        """Init method."""
+        super().__init__(args)
+
+    def process(self):
+        pass # TBD
+
+class RestoreCmd(Cmd):
+
+    """Restore Cmd"""
+    name = "restore"
+
+    def __init__(self, args):
+        """Init method."""
+        super().__init__(args)
+
+    def process(self):
+        pass # TBD
+
+def main(argv: dict):
+    try:
+        desc = "HA Setup command"
+        command = Cmd.get_command(desc, argv[1:])
+        command.process()
+
+    except Exception as e:
+        sys.stderr.write("error: %s\n\n" % str(e))
+        sys.stderr.write("%s\n" % traceback.format_exc())
+        Cmd.usage(argv[0])
+        return errno.EINVAL
+
+if __name__ == '__main__':
+    Conf.init(delim='.')
+    Conf.load(const.HA_GLOBAL_INDEX, f"yaml://{const.HA_CONFIG_FILE}")
```

Comment:
```
    Conf.init(delim='.')
    Conf.load(const.HA_GLOBAL_INDEX, f"yaml://{const.HA_CONFIG_FILE}")
    log_path = Conf.get(const.HA_GLOBAL_INDEX, "LOG.path")
    log_level = Conf.get(const.HA_GLOBAL_INDEX, "LOG.level")
    Log.init(service_name='ha_setup', log_path=log_path, level=log_level)
```
can you create file 
ha/core/config/config_manager.py 
like https://github.com/Seagate/cortx-ha/pull/279/files

to get configuration
it is needed by many file and using config_manager will avoid duplicate code

Reply:
I think lets reuse after PR #279 gets merged. I will add todo for the same.